test/client/fixtures/redirect-watch-test.js

```diff
@@ -177,85 +180,153 @@ asyncTest('location.search = ...', function () {
-        .then(function () {
-            var iframeHammerhead = iframe.contentWindow['%hammerhead%'];
+asyncTest('Submit form by submit button click', function () {
+    var redirectScript = 'var form = document.createElement("form");' +
+                         'var submit = document.createElement("input");' +
+                         'form.action = "./index.html";' +
+                         'submit.type = "submit";' +
+                         'form.appendChild(submit);' +
+                         'document.body.appendChild(form);' +
+                         'submit.click();';
 
-            iframeHammerhead.on(iframeHammerhead.EVENTS.redirectDetected, function (e) {
-                strictEqual(e, iframeLocation + 'index.html');
-                document.body.removeChild(iframe);
-                start();
-            });
+    redirectIframe(redirectScript).then(function (e) {
+        strictEqual(e, iframeLocation + 'index.html');
+        start();
+    });
+});
 
-            iframe.contentWindow.eval(
-                'var link = document.createElement("a");' +
-                'link.href = "./index.html";' +
-                'document.body.appendChild(link);' +
-                'link.click();'
-            );
-        });
+asyncTest('Submit form via js', function () {
+    var redirectScript = 'var form = document.createElement("form");' +
+                         'form.action = "./index.html";' +
+                         'document.body.appendChild(form);' +
+                         'form.submit();';
 
-    document.body.appendChild(iframe);
+    redirectIframe(redirectScript).then(function (e) {
+        strictEqual(e, iframeLocation + 'index.html');
+        start();
+    });
 });
 
-if (!browserUtils.isWebKit) {
-    asyncTest('Link with target attribute', function () {
-        var iframe = document.createElement('iframe');
+asyncTest('Submit form with the target attribute', function () {
+    var redirectScript = 'var form = window.top.document.createElement("form");' +
+                         'form.setAttribute("action", location.toString() + "index.html");' +
+                         'form.setAttribute("target", "submitIframe");' +
+                         'window.top.document.body.appendChild(form);' +
+                         'form.submit();';
 
-        iframe.id   = 'test' + Date.now();
-        iframe.name = 'iframeName';
-        iframe.src  = location.protocol + '//' + location.host + '/unchangeableUrlSession!i/' + iframeLocation;
+    redirectIframe(redirectScript, true, 'submitIframe').then(function (e) {
+        strictEqual(e, iframeLocation + 'index.html');
+        start();
+    });
+});
 
-        window.QUnitGlobals.waitForIframe(iframe)
-            .then(function () {
-                var iframeHammerhead = iframe.contentWindow['%hammerhead%'];
+asyncTest('Submission canceled in the "addEventListener" method', function () {
+    var redirectScript = 'var form = document.createElement("form");' +
```

Comment:
I have investigated, this is a global problem, I fix this in the next pull request.